doc/nrf/libraries/modem/nrf_modem_lib.rst

```diff
@@ -23,6 +23,10 @@ The library wrapper eases the task of initializing the Modem library by automati
 :ref:`partition_manager` is the component that reserves the RAM memory for the shared memory regions used by the Modem library.
 For more information, see :ref:`partition_mgr_integration`.
 
+The library wrapper also provides callbacks for the initialization and shutdown operations.
+The application can setup a callback for :c:func:`nrf_modem_lib_init` calls using the :c:macro:`NRF_MODEM_LIB_ON_INIT` macro, and a callback for :c:func:`nrf_modem_lib_shutdown` calls using the :c:macro:`NRF_MODEM_LIB_ON_SHUTDOWN` macro.
+These compile-time callbacks allow any part of the application to perform any setup steps that require the modem to be a certain state. Futher, they ensure that said steps will be repeated any time another part of the application turns the modem on or off.
```

Comment:
Not sure how relevant it is, but should this specify that the init callbacks are called after the modem is initialized and the shutdown callbacks are called before the modem is shut down?

Reply:
Good point, I'll mention it here and in the doxygen.